### ISSUE TITLE
Idea: visual indicator if a weakness is 4x effective 

### ISSUE THREAD

**Issue by ArchOwlen** [roguedex-dev/roguedex#17]
Not something I would use since I know typings, but could be useful for new pokemon players' ease-of-use

Also not sure what would make it "pop" the best without making a new box, as say: a "blue border" around the icon (like a discord status circle around the icon) wouldn't work well for blue-colored typings like ice (color blindness in mind especially)

Perhaps a 4x sub-circle to that weakness?
Though it may also be hard to see when small

**Comment by ArchOwlen** [comment]
This could also apply to 4x resistance, but the implementation could probably be the same as the type  being resisted would be in the "strength" box, so context could be used to determine the 4x means resist 

(In other words, the 4x sub-circle could be used in both strength and weakness as it would already be in the strength/weakness zone)

**Comment by roguedex-dev** [comment]
The problem with this is that via PokeAPI I have no way of knowing if a type is 2x or 4x weak/resistant to something.

**Comment by aalejandrofer** [comment]
You will have to calculate it. Let's say you have a Fighting and Bug because both have Flying in the 2x table then Flying becomes 4x. 

Maybe something like this.

```js
async function isDoubleWeak(pokemonName, attackType) {
    const response = await fetch(`https://pokeapi.co/api/v2/pokemon/${pokemonName}`);
    const data = await response.json();

    const pokemonTypes = data.types.map(typeData => typeData.type.name);
    for (let pokemonType of pokemonTypes) {
        const typeInfo = await getTypeInfo(pokemonType);
        if (typeInfo.double_damage_from.some(type => type.name === attackType)) {
            return true;
        }
    }
    return false;
}
```

**Comment by aalejandrofer** [comment]
It could be the same for 0.5x. I also like the idea of just doing a ring around the logo or something like that.

**Comment by roguedex-dev** [comment]
So what you're saying is that there is no single type with 4x weakness / 1/4x resistance, and that it always comes from double typing?

**Comment by aalejandrofer** [comment]
To my understanding yes. Look at this site (https://pkmn.help/) is also open-source.

**Comment by ArchOwlen** [comment]
In 99% of circumstances, yes, it is only possible with double typing 

However, there are a few edge cases introduced in Scarlet/Violet where a move's description states 
"This move is more effective against [type/s]"

I don't recall the specific move names, but I did spot one being used in PokeRogue 
However, I don't remember if it was partially implemented as of right now

**Comment by aalejandrofer** [comment]
Salt Cure is an example I believe of that.

Salt Cure inflicts 1/8 of the target's maximum HP as damage per turn in addition to the damage dealt when it is used. If a [Steel](https://bulbapedia.bulbagarden.net/wiki/Steel_(type)) and/or [Water](https://bulbapedia.bulbagarden.net/wiki/Water_(type)) type is affected by Salt Cure, the amount of damage per turn is ¼ of its maximum HP.

**Comment by aalejandrofer** [comment]
Also to my point above about DNA Splitting, also Terraforming affects what a pokemon is weak too. Thats why if possible I suggest only looking at the typing of the currentEnemy rather than the pokemon name/id.

**Comment by ArchOwlen** [comment]
Yep, just tested and made #28 (even though it may be considered redundant)

We should probably track types rather than the pokemon themselves
